tools.sh

```diff
@@ -128,5 +139,16 @@ function start_vm {
     done
 }
 
+function generate_htpasswd_file {
+   local auth_file_dir=$1
+   local pass_file=$2
+   random_password=$(${OPENSSL} rand -base64 14)
+   ${HTPASSWD} -c -B -b ${pass_file} developer developer
+   ${HTPASSWD} -B -b ${pass_file} kubeadmin ${random_password}
+
+   # Update kubeadmin-password file
+   echo ${random_password} > $1/auth/kubeadmin-password
```

Comment:
The newline at the end of the password is unexpected. crc takes the \n has a character of the password. 

I guess you need to use `echo -n`.

Reply:
I opened this https://github.com/code-ready/crc/pull/2046, it also fixes the problem.

Reply:
@guillaumerose sure we can control the way of putting the password in the file.